lib/opentsdb_consumer.rb

```diff
@@ -0,0 +1,54 @@
+module OpenTSDBConsumer
+  def self.fetch(host: '', port: '', metric_names: [], metric_options: {})
+    Facade.new(host, port, metric_names).fetch(metric_options)
+  end
+
+  class Facade
+    def initialize(host, port, metric_names)
+      @client       = Client.new(host: host, port: port)
+      @metric_names = metric_names
+    end
+
+    def fetch(metric_options = {})
+      @metric_options = {
+        start: '8h-ago',
+        aggregator: 'sum',
+        downsample: '5m-avg',
+        rate: false,
+        tags: nil,
+      }.merge(metric_options)
+
+      @data ||= query.run(start: metric_options[:start])
+    rescue InvalidTag, InvalidMetric
+      OpenStruct.new datapoints: []
+    end
+
+    private
+
+    attr_reader :client, :metric_names, :metric_options
```

Comment:
It's a matter of style obviously but It's to avoid having a lot of `@` everywhere, which I and some ruby members don't like :).